cel/runtimecost_test.go

```diff
@@ -0,0 +1,433 @@
+package cel
+
+import (
+	"github.com/google/cel-go/checker"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/interpreter"
+	"github.com/google/cel-go/test/proto3pb"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+	"testing"
+	"time"
+)
+
+type testInfo struct {
+	env     *Env
+	in      interface{}
+	lhsExpr string
+	rhsExpr string
+}
+
+//func computeCosts(t *testing.T, info *testInfo) (lhsCost, rhsCost uint64) {
+//	t.Helper()
+//
+//	env := info.env
+//	if env == nil {
+//		emptyEnv, err := NewEnv()
+//		if err != nil {
+//			t.Fatalf("Failed to create empty environment, error: %v", err)
+//		}
+//		env = emptyEnv
+//	}
+//	ctx := constructActivation(t, info.in)
+//	lhsCost = computeCost(t, env, info.lhsExpr, &ctx)
+//	rhsCost = computeCost(t, env, info.rhsExpr, &ctx)
+//
+//	return lhsCost, rhsCost
+//}
+
+//func TestTrackCostAdvanced(t *testing.T) {
+//	var equalCases = []testInfo{
+//		{
+//			lhsExpr: `1`,
+//			rhsExpr: `2`,
+//		},
+//		{
+//			lhsExpr: `"abc".contains("d")`,
+//			rhsExpr: `"def".contains("d")`,
+//		},
+//		{
+//			lhsExpr: `1 in [4, 5, 6]`,
+//			rhsExpr: `2 in [15, 17, 16]`,
+//		},
+//	}
+//	for i, testCase := range equalCases {
+//		lhsCost, rhsCost := computeCosts(t, &testCase)
+//		if lhsCost != rhsCost {
+//			t.Errorf(`Expected equal cost case #%d, expressions "%s" vs. "%s", respective cost %d vs. %d`, i,
+//				testCase.lhsExpr, testCase.rhsExpr, lhsCost, rhsCost)
+//		}
+//	}
+//	var smallerCases = []testInfo{
+//		{
+//			lhsExpr: `1`,
+//			rhsExpr: `1 + 2`,
+//		},
+//		{
+//			lhsExpr: `"abc".contains("d")`,
+//			rhsExpr: `"abcd".contains("e")`,
+//		},
+//		{
+//			lhsExpr: `1 in [4, 5, 6]`,
+//			rhsExpr: `1 in [4, 5, 6, 7, 8, 9]`,
+//		},
+//	}
+//	for i, testCase := range smallerCases {
+//		lhsCost, rhsCost := computeCosts(t, &testCase)
+//		if lhsCost >= rhsCost {
+//			t.Errorf(`Expected smaller cost case #%d, expect the cost of expression "%s" to be strictly smaller than "%s", but got %d vs. %d respectively`,
+//				i, testCase.lhsExpr, testCase.rhsExpr, lhsCost, rhsCost)
+//		}
+//	}
+//}
+
+func TestRuntimeCost(t *testing.T) {
```

Comment:
I'd strongly prefer we don't duplicate all the tests cases from estimated cost.  Instead could we have the estimated cost test suite covers both estimated and actual cost (renaming it as appropriate)?